server/src/controllers/eventController.ts

```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    return res.status(500).json({error: 'Internal Server Error'});
+  }
+});
+
+/**
+ * @swagger
+ * /2/events/search:
+ *   get:
+ *     summary: Search events by name
+ *     tags:
+ *     - New Event API
+ *     parameters:
+ *       - in: query
+ *         name: event_name
+ *         schema:
+ *           type: string
+ *     security:
+ *       - bearerAuth: []
+ *     responses:
+ *       200:
+ *         description: A list of event IDs and names matching the provided event name
+ *       400:
+ *         description: bad request
+ *       500:
+ *         description: An error occurred while processing the request
+ */
+eventController.get('/search', async (req: Request, res: Response) => {
+  try {
+    const name = req.query.event_name;
+    if (!name || !(typeof name === 'string')) {
+      return res.status(400).send('No Event name provided.');
+    }
+
+    const events = await prisma.events.findMany({
+      where: {
+        eventname: name,
+      },
+    });
+    return res.json(events);
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    return res.status(500).json({error: 'Internal Server Error'});
+  }
+});
+
+/**
+ * @swagger
+ * /2/events/{id}:
+ *   get:
+ *     summary: get an event do not include showings
+ *     tags:
+ *     - New Event API
+ *     parameters:
+ *     - $ref: '#/components/parameters/id'
+ *     security:
+ *       - bearerAuth: []
+ *     responses:
+ *       200:
+ *         description: event fetch successful.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Event'
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+eventController.get('/:id', async (req: Request, res: Response) => {
+  try {
+    const id = req.params.id;
+    const eventExists = await prisma.events.findUnique({
+      where: {
+        eventid: Number(id),
+      },
+    });
+    if (!eventExists) {
+      res.status(400).json({error: `Event ${id} not found`});
+      return;
+    }
+    res.status(200).json(eventExists);
+    return;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(400).json({error: error.message});
+
```

Comment:
Remove newlines like this from here down

Reply:
fixed